### PR TITLE
WIP 5684 loan response validation

### PR DESCRIPTION
**Pull request by X0nic**

### CODE REVIEW

File: app/models/loan_response.rb [hunk 1]
```diff
@@ -120,6 +123,10 @@ def required?
 
   private
 
+  def breakeven_table_requires_name
+    errors.add(:breakeven_table, 'All products require a name') if breakeven_hash[:products].any?{ |p| p[:name].blank? }
```
Comment (linesarefuzzy) [inline app/models/loan_response.rb]: I18n

File: app/models/loan_response_set.rb [hunk 2]
```diff
@@ -33,16 +33,32 @@ def recalculate_loan_health
     RecalculateLoanHealthJob.perform_later(loan_id: loan_id)
   end
 
+  validate do
+    # iterate through all LoanResponses and set a :base error if any of them are invalid
+    errors.add(:base, 'Validation error') unless responses.all?(&:valid?)
```
Comment (linesarefuzzy) [inline app/models/loan_response_set.rb]: I18n

File: app/models/loan_response_set.rb [hunk 3]
```diff
@@ -74,19 +90,21 @@ def active?
...
-
   # Fetches a custom value from the json field. `question_identifier` can be the same
   def response(question_identifier)
     field = question(question_identifier)
     raw_value = (custom_data || {})[field.json_key]
+    # Here, instead of just blindly creating a new one, check the hash first.
     LoanResponse.new(loan: loan, loan_question: field, loan_response_set: self, data: raw_value)
   end
 
+  # This seems like a wonkey way to do it. Is there a better way?
```
Comment (linesarefuzzy) [inline app/models/loan_response_set.rb]: Maybe something like
```
custom_data.each do |key, val|
    LoanResponse.new(loan: loan, loan_question: question(key), loan_response_set: self, data: val)
end
```
But I don't know if that would solve your problem.

**Comment by emitche** [comment]
I'm closing this branch, as there has not been movement for several weeks. Please reopen this branch or start a new one in the future. Thank you.

**Pull request closed without merging**